File: IT/Oggetti/Python/polym.py
# file polym.py

from __future__ import print_function

# dichiara il campo "age" nella superclasse Animal
class Animal(object):
	def __init__(self, age):
		self.age = age

	def getAge(self):
		return self.__age

	def setAge(self, age):
		self.__age = age

	age = property(getAge, setAge)

# la classe Person estende la classe Animal ed eredita i membri

class Person(Animal):
	def __init__(self, name, surname, age):
		super(Person, self).__init__(age)
		self.__name = name
		self.__surname = surname

	def __str__(self):
		return "{} {}, {} anni.".format(self.__name, self.__surname, self.age)

# crea un'istanza della sottoclasse Person

p = Person("John", "Doe", 30)

# verifica che p appartenga al tipo Person, ...ma anche al tipo Animal

print("p animale? {}".format(isinstance(p, Animal)))
print("p persona? {}".format(isinstance(p, Person)))

# stampa l'oggetto con il metodo __str__()
print(p)

File: IT/Oggetti/Java/Polym.java
/* file Polym.java */

/* dichiara il campo "age" nella superclasse Animal */
class Animal {
	private int age;

	public Animal(int age) {
		setAge(age);
	}

	public int getAge() {
		return this.age;
	}

	public void setAge(int age) {
		this.age = age;
	}
}

/* la classe Person estende la classe Animal ed eredita i suoi membri */

class Person extends Animal {
	private String name;
	private String surname;

	public Person(String name, String surname, int age) {
		super(age);
		this.setName(name);
		this.setSurname(surname);
	}

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getSurname() {
		return this.surname;
	}

	public void setSurname(String surname) {
		this.surname = surname;
	}

	@Override
	public String toString() {
		return this.name + " " + this.surname + ", " + this.getAge() + " anni.";
	}
}

class Env {
	public static void main(String [] args) {
		/* crea un'istanza della sottoclasse Person */

		Person p = new Person("John", "Doe", 30);

		/* verifica che p appartenga al tipo Person, ...ma anche al tipo Animal */

		System.out.println("p animale? " + (p instanceof Animal));
		System.out.println("p persona? " + (p instanceof Person));

		// stampa l'oggetto con il metodo toString()
		System.out.println(p);
	}
}

